app/uk/gov/hmrc/excisemovementcontrolsystemapi/services/MessageService.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.excisemovementcontrolsystemapi.services
+
+import uk.gov.hmrc.excisemovementcontrolsystemapi.models.MessageTypes.{IE801, IE810, IE815, IE818, IE837}
+import uk.gov.hmrc.excisemovementcontrolsystemapi.models.messages._
+import uk.gov.hmrc.excisemovementcontrolsystemapi.repository.MovementRepository
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class MessageService @Inject()(movementRepository: MovementRepository, implicit val executionContext: ExecutionContext) {
+
+  private def getErnsUsingArc(arc: String): Future[Set[String]] = {
```

Comment:
what about somethiong like that which seem for functional? And may more readable as you just deal with one future only?

` movementRepository.getMovementByARC(arc).map {
      case Seq() => throw new RuntimeException(s"[MessageService] - Zero movements found for administrative reference code $arc")
      case head :: Nil => Set(Some(head.consignorId), head.consigneeId).flatten
      case _ => throw new RuntimeException(s"[MessageService] - Multiple movements found for administrative reference code $arc")
    }`